src/AppInstallerRepositoryCore/Public/winget/RepositorySource.h

```diff
@@ -0,0 +1,249 @@
+        std::vector<SourceAgreement> SourceAgreements;
+
+        // Unsupported match fields in search request. If this field is in the filters, the request may fail.
+        std::vector<std::string> UnsupportedPackageMatchFields;
+
+        // Required match fields in search request. If this field is not found in the filters, the request may fail(except Market).
+        std::vector<std::string> RequiredPackageMatchFields;
+
+        // Unsupported query parameters in get manifest request.
+        std::vector<std::string> UnsupportedQueryParameters;
+
+        // Required query parameters in get manifest request.
+        std::vector<std::string> RequiredQueryParameters;
+    };
+
+    // Represents a source which would be interacted from outside of repository lib.
+    struct Source
+    {
+        // Default constructor to get all sources available. Can be used in source list, etc.
+        Source();
+
+        // Constructor to get a named source, passing empty string will get all available sources(same as the default constructor).
+        Source(std::string_view name);
+
+        // Constructor to get a PredefinedSource. Like installed source, etc.
+        Source(PredefinedSource source);
+
+        // Constructor to get a source coming with winget. Like winget community source, etc.
+        Source(WellKnownSource source);
+
+        // Constructor for a source to be added.
+        Source(std::string_view name, std::string_view arg, std::string_view type);
+
+        // Constructor for creating a composite source from a list of available sources.
+        Source(const std::vector<Source>& availableSources);
+
+        // Constructor for creating a composite source from an installed source and available source(may be composite already).
+        Source(
+            const Source& installedSource,
+            const Source& availableSource,
+            CompositeSearchBehavior searchBehavior = CompositeSearchBehavior::Installed);
+
+        // Bool operator to check if a source reference is successfully acquired.
+        // Theoretically, the constructor could just throw when CreateSource returns empty.
+        // To avoid putting try catch everywhere, we use bool operator here.
+        operator bool() const;
+
+        // Gets the source's identifier; a unique identifier independent of the name
+        // that will not change between a remove/add or between additional adds.
+        // Must be suitable for filesystem names unless the source is internal to winget,
+        // in which case the identifier should begin with a '*' character.
+        const std::string GetIdentifier() const;
+
+        // Get the source's configuration details from settings.
+        const SourceDetails GetDetails() const;
+
+        // Get the source's information.
+        const SourceInformation GetInformation() const;
+
+        // Set custom header.
+        bool SetCustomHeader(std::string_view header);

```

Comment:
```suggestion
        bool SetCustomHeader(std::optional<std::string_view> header);
```

Or offer a function to clear it. #Resolved